solr/core/src/java/org/apache/solr/cluster/placement/impl/PlacementRequestImpl.java

```diff
@@ -26,29 +25,23 @@
 import org.apache.solr.cluster.Replica;
 import org.apache.solr.cluster.SolrCollection;
 import org.apache.solr.cluster.placement.PlacementRequest;
+import org.apache.solr.common.cloud.ReplicaCount;
 
 public class PlacementRequestImpl implements PlacementRequest {
   private final SolrCollection solrCollection;
   private final Set<String> shardNames;
   private final Set<Node> targetNodes;
-  private final EnumMap<Replica.ReplicaType, Integer> countReplicas =
-      new EnumMap<>(Replica.ReplicaType.class);
+  private final ReplicaCount numReplicas;
 
   public PlacementRequestImpl(
       SolrCollection solrCollection,
       Set<String> shardNames,
       Set<Node> targetNodes,
-      int countNrtReplicas,
-      int countTlogReplicas,
-      int countPullReplicas) {
+      ReplicaCount numReplicas) {
```

Comment:
Not sure if it's worth adding an overloaded variant for backwards-compatibility, albeit marked deprecated and a comment to remove in 10?

Reply:
Sure, fixed.